src/uwtools/drivers/forecast.py

```diff
@@ -40,33 +49,74 @@ def validate(self):
 
     def create_model_config(self):
 
-        ''' Collects all the user inputs required to create a model config 
+        ''' Collects all the user inputs required to create a model config
         file, calling the existing model config tools. This will be unique
         to the app being run and will appropriately parse subsequent stages
-        of the workflow. 
-        Defaults will be filled in if not provided by the user. Equivalent 
+        of the workflow.
+        Defaults will be filled in if not provided by the user. Equivalent
         references to config_default.yaml or config.community.yaml from SRW
         will need to be made for the other apps. '''
 
     def stage_fix_files(self):
 
-        ''' Holds the knowledge for how to modify a list of fix files and 
-        stages them in the working directory. Likely gets all its info from 
-        config_obj. Calls data mover tool (could be python copy). Fix files 
+        ''' Holds the knowledge for how to modify a list of fix files and
+        stages them in the working directory. Likely gets all its info from
+        config_obj. Calls data mover tool (could be python copy). Fix files
         usually are specific to a given named grid and resolution. '''
 
     def create_namelist(self):
-        ''' Collects all the user inputs required to create a namelist 
+        ''' Collects all the user inputs required to create a namelist
         file, calling the existing namelist config tools. For SRW,
         this will take the fix files and model_config previously created to
         to generate config.yaml'''
 
+    def create_directory_structure(self, run_directory, exist_act="delete"):
```

Comment:
Sorry I haven't caught this question sooner. Does it make sense for run_directory to be an argument to this method, or for it to be an object attribute?

Reply:
@christinaholtNOAA UW-12 says, "The location of that directory will be communicated to the script by the caller." Are there OO rules of thumb that suggest when a piece of information should be an attribute vs an argument? Is this something that an earlier method will have found in a schema file and want to squirrel away into an attribute?

Reply:
In general, I think about the OO principle such that an object is a set of data with a group of actions associated with it. If a piece of data is inherently tied to the object then it might be helpful to associate that data to the specific object. So, in this case, if I decided to create a whole set of forecast objects for an experiment (one every 6 hours, maybe), then each of those objects could tell me what location on disk it was configured to run.

I could see how we might just have a configuration object associated with each Forecast object and then query tht to get the same information. I have no rule of thumb to decide between those two choices.

Now that I've written it out like this...let's keep it as-is. If we choose to make it an attribute later, along with any of the other configuration settings required by its configuration object, we can refactor.